relay-profiling/src/android/legacy.rs

```
@@ -236,9 +236,12 @@ pub fn parse_android_profile(
         environment.clone_into(&mut profile.metadata.environment);
     }
 
-    if let Some(segment_id) = transaction_metadata.get("segment_id") {
+    if let Some(segment_id) = transaction_metadata
+        .get("segment_id")
+        .and_then(|segment_id| segment_id.parse().ok())
+    {
```

Is it ok to silently drop invalid segment IDs here?

---

I think that makes sense.